### PR TITLE
Flow Graph Context Logger

### PR DESCRIPTION
**Pull request by carolhmj**
# Changes
- Add a new observable in the context, `onNodeExecutedObservable`, which is notified every time a node is executed. This observable can be subscribed to for various functionalities, such as logging
- Add the FlowGraphContextLogger, which is subscribed to the observable above and logs when each node is executed.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/FlowGraph/flowGraphContextLogger.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type { FlowGraphContext } from "./flowGraphContext";
+
+/**
+ * @experimental
+ * This class is a decorator for the context which logs the nodes that were executed.
+ */
+export class FlowGraphContextLogger {
+    constructor(private _context: FlowGraphContext) {
+        this._context.onNodeExecutedObservable.add((node) => {
+            console.log(`Node executed: ${node.getClassName()}`);
```
Comment (RaananW) [inline packages/dev/core/src/FlowGraph/flowGraphContextLogger.ts:10]: we should use the logger instead of console.log

**Review by RaananW — APPROVED** [review]
That's a great debugging tool. Is there any use for it other than during development?

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14495/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14495/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14495/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14495/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14495/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14495/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14495/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14495/merge#BCU1XR#0

**Pull request merged**